source/common/network/listener_filter_buffer_impl.cc

```diff
@@ -16,9 +16,18 @@ ListenerFilterBufferImpl::ListenerFilterBufferImpl(IoHandle& io_handle,
   // If the buffer_size not greater than 0, it means that doesn't expect any data.
   ASSERT(buffer_size > 0);
 
+// `FileReadyType::Closed` is required `EV_FEATURE_EARLY_CLOSE` feature for libevent, and this feature is
+// only supported with `epoll`. But `MacOS` uses the `kqueue`.
+// https://libevent.org/doc/event_8h.html#a98f643f9c9063a4cbf410f519eb61e55
+#if defined(__APPLE__)
```

Comment:
@ggreenway I'm a little hesitant to add this check, even if we add `Closed` event for macOS, it won't break anything. maybe without this check will make the code clear? We can just skip the test for macOS. but let me know your opinion.

Reply:
I think the check can be moved to the test only. Setting the Closed event doesn't cause any harm on macos I don't think; it just doesn't do anything.

Reply:
got it